NoClassDefFoundError when selecting spring-data-jdbc feature

### Description

The Micronaut app generated with the spring-data-jdbc feature fails to run due to
  `java.lang.NoClassDefFoundError: io/micronaut/transaction/hibernate5/MicronautSessionContext`

### Steps to Reproduce

1. Configure a Java 11 starter app with gradle and spring-data-jdbc: https://micronaut.io/launch?type=DEFAULT&name=demo-jdbc&package=com.example&javaVersion=JDK_11&lang=JAVA&build=GRADLE&test=JUNIT&features=spring-data-jdbc&version=3.7.1
2. Unzip it.
3. Attempt to run it via ./gradlew run

### Expected Behaviour

It should start w/o exceptions.

### Actual Behaviour

Receive an exception ending with:
`Caused by: java.lang.ClassNotFoundException: io.micronaut.transaction.hibernate5.MicronautSessionContext
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)`

### Environment Information

- **Operating System**: Ubuntu under WSL
- **Micronaut Version:** 3.7.1
- **JDK Version:** 11

### Example Application

https://micronaut.io/launch?type=DEFAULT&name=demo-jdbc&package=com.example&javaVersion=JDK_11&lang=JAVA&build=GRADLE&test=JUNIT&features=spring-data-jdbc&version=3.7.1

--- a/starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/MicronautApplicationGradlePlugin.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/MicronautApplicationGradlePlugin.java
@@ -51,6 +51,7 @@ public class MicronautApplicationGradlePlugin {
         private boolean incremental;
         private  String packageName;
         private boolean sharedTestResources;
+
         public Builder buildTool(BuildTool buildTool) {
             this.buildTool = buildTool;
             return this;

--- a/starter-core/src/main/java/io/micronaut/starter/feature/spring/SpringDataJdbc.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/spring/SpringDataJdbc.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017-2022 original authors
+ * Copyright 2017-2023 original authors
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -18,6 +18,7 @@ package io.micronaut.starter.feature.spring;
 import io.micronaut.starter.application.ApplicationType;
 import io.micronaut.starter.application.generator.GeneratorContext;
 import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.dependencies.MicronautDependencyUtils;
 import io.micronaut.starter.feature.FeatureContext;
 import io.micronaut.starter.feature.database.DataJdbc;
 
@@ -25,6 +26,21 @@ import jakarta.inject.Singleton;
 
 @Singleton
 public class SpringDataJdbc extends SpringFeature {
+
+    private static final Dependency MICRONAUT_DATA_SPRING_DEPENDENCY = MicronautDependencyUtils
+            .dataDependency()
+            .artifactId("micronaut-data-spring")
+            .compile()
+            // exclude for Micronaut 3.x only, fixed in Micronaut 4.x
+            .exclude(MicronautDependencyUtils.sqlDependency().artifactId("micronaut-hibernate-jpa").build())
+            .exclude(MicronautDependencyUtils.sqlDependency().artifactId("micronaut-hibernate-jpa-spring").build())
+            .build();
+
+    private static final Dependency SPRING_JDBC_DEPENDENCY = Dependency.builder()
+            .groupId("org.springframework")
+            .artifactId("spring-jdbc")
+            .compile()
+            .build();
 
     private final DataJdbc dataJdbc;
 
@@ -63,13 +79,7 @@ public class SpringDataJdbc extends SpringFeature {
 
     @Override
     public void apply(GeneratorContext generatorContext) {
-        generatorContext.addDependency(Dependency.builder()
-                .groupId("io.micronaut.data")
-                .artifactId("micronaut-data-spring")
-                .compile());
-        generatorContext.addDependency(Dependency.builder()
-                .groupId("org.springframework")
-                .artifactId("spring-jdbc")
-                .compile());
+        generatorContext.addDependency(MICRONAUT_DATA_SPRING_DEPENDENCY);
+        generatorContext.addDependency(SPRING_JDBC_DEPENDENCY);
     }
 }

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/spring/SpringDataJdbcSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/spring/SpringDataJdbcSpec.groovy
@@ -3,6 +3,9 @@ package io.micronaut.starter.feature.spring
 import io.micronaut.starter.ApplicationContextSpec
 import io.micronaut.starter.BuildBuilder
 import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.build.BuildTestUtil
+import io.micronaut.starter.build.BuildTestVerifier
+import io.micronaut.starter.build.dependencies.Scope
 import io.micronaut.starter.feature.Category
 import io.micronaut.starter.feature.Features
 import io.micronaut.starter.options.BuildTool
@@ -53,112 +56,23 @@ class SpringDataJdbcSpec extends ApplicationContextSpec {
     }
 
     @Unroll
-    void 'test spring-data-jdbc with Gradle for language=#language'() {
+    void 'test spring-data-jdbc with buildTool=#buildTool for language=#language'(Language language, BuildTool buildTool) {
         when:
-        String template = new BuildBuilder(beanContext, BuildTool.GRADLE)
+        String template = new BuildBuilder(beanContext, buildTool)
                 .features(['spring-data-jdbc'])
                 .language(language)
                 .render()
 
+        BuildTestVerifier verifier = BuildTestUtil.verifier(buildTool, language, template)
+
         then:
-        template.contains("${getGradleAnnotationProcessorScope(language)}(\"io.micronaut.spring:micronaut-spring-annotation\")")
-        template.contains('implementation("io.micronaut.data:micronaut-data-spring")')
-        template.contains('implementation("org.springframework:spring-jdbc")')
+        verifier.hasDependency("io.micronaut.data", "micronaut-data-spring")
+        verifier.hasExclusion("io.micronaut.sql", "micronaut-hibernate-jpa")
+        verifier.hasExclusion("io.micronaut.sql", "micronaut-hibernate-jpa-spring")
+        verifier.hasDependency("org.springframework", "spring-jdbc")
+        verifier.hasAnnotationProcessor("io.micronaut.spring", "micronaut-spring-annotation")
 
         where:
-        language << Language.values().toList()
-    }
-
-    void 'test maven spring-data-jdbc feature for java'() {
-        when:
-        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
-                .features(['spring-data-jdbc'])
-                .language(Language.JAVA)
-                .render()
-
-        then:
-        template.contains("""
-    <dependency>
-      <groupId>io.micronaut.data</groupId>
-      <artifactId>micronaut-data-spring</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-    <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-jdbc</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-            <path>
-              <groupId>io.micronaut.spring</groupId>
-              <artifactId>micronaut-spring-annotation</artifactId>
-              <version>\${micronaut.spring.version}</version>
-            </path>
-""")
-    }
-
-    void 'test maven spring-data-jdbc feature for groovy'() {
-        when:
-        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
-                .features(['spring-data-jdbc'])
-                .language(Language.GROOVY)
-                .render()
-
-        then:
-        template.contains("""
-    <dependency>
-      <groupId>io.micronaut.data</groupId>
-      <artifactId>micronaut-data-spring</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-    <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-jdbc</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-    <dependency>
-      <groupId>io.micronaut.spring</groupId>
-      <artifactId>micronaut-spring-annotation</artifactId>
-      <scope>provided</scope>
-    </dependency>
-""")
-    }
-
-    void 'test maven spring-data-jdbc feature for kotlin'() {
-        when:
-        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
-                .features(['spring-data-jdbc'])
-                .language(Language.KOTLIN)
-                .render()
-
-        then:
-        template.contains("""
-    <dependency>
-      <groupId>io.micronaut.data</groupId>
-      <artifactId>micronaut-data-spring</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-    <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-jdbc</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-               <annotationProcessorPath>
-                 <groupId>io.micronaut.spring</groupId>
-                 <artifactId>micronaut-spring-annotation</artifactId>
-                 <version>\${micronaut.spring.version}</version>
-               </annotationProcessorPath>
-""")
+        [language, buildTool] << [Language.values(), BuildTool.values()].combinations()
     }
 }

